integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala

```diff
@@ -83,7 +87,7 @@ object IndexServer extends ServerInterface {
   private val isExecutorLRUConfigured: Boolean =
     CarbonProperties.getInstance
       .getProperty(CarbonCommonConstants.CARBON_MAX_EXECUTOR_LRU_CACHE_SIZE) != null
-
+  private val operationContext: OperationContext = new OperationContext
```

Comment:
This operation context would be common for all the calls to index server. I think its safe to make this as a local variable wherever needed